tests/core/test_core.py

```diff
@@ -43,6 +43,23 @@ def test_initialize_controller(self, controller, mocker) -> None:
         self.model.poll_for_events.assert_called_once_with()
         assert controller.theme == self.theme
 
+    def test_initial_editor_mode(self, controller):
```

Comment:
Maybe extract this change along with the core.py changes into a previous commit?

Reply:
These tests are for the new functions, which replace the previous direct setting of the attributes. Do you mean to add the new methods and tests in one commit, and then migrate the code to use them in another? I suppose I could make the attributes private in the second step.

Reply:
Yes, exactly.